nmigen/hdl/rec.py

```diff
@@ -146,7 +144,18 @@ def concat(a, b):
                                                      src_loc_at=1 + src_loc_at)
 
     def __getattr__(self, name):
-        return self[name]
+        # must check `getattr` before `self` - we need to hit Value methods before fields
+        try:
+            # it seems we lose the implicit `self` argument at some point, so curry manually
```

Comment:
Python adds a `self` argument on its own in exactly one (AFAIK) case: when an attribute access on an *object* resolves to an attribute of its *class* and this attribute is *callable*. Since in this case you are accessing a class attribute, not an object attribute, you simply fetch the function from the class unmodified:
```pyrepl
>>> class c:
...     def f(x,y): return x,y
...
>>> c.f
<function c.f at 0x7fb25aeafe18>
>>> c.f(1,2)
(1, 2)
```
There's nothing special about the code above--the class acts as a fancy dictionary.

Reply:
To make sure I understand what you're saying - I'm doing the right thing here, but the comment displaying my confusion is unnecessary?

Reply:
More or less.